modules/mbedtls/Kconfig.tls-generic

```diff
@@ -97,42 +97,40 @@ config MBEDTLS_KEY_EXCHANGE_DHE_RSA_ENABLED
 
 config MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED
 	bool "ECDHE-RSA based ciphersuite modes"
-	select MBEDTLS_ECDH_C
+	depends on MBEDTLS_ECDH_C
 
 config MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED
 	bool "ECDHE-ECDSA based ciphersuite modes"
-	select MBEDTLS_ECDH_C
-	select MBEDTLS_ECDSA_C
+	depends on MBEDTLS_ECDH_C && MBEDTLS_ECDSA_C
 
 config MBEDTLS_KEY_EXCHANGE_ECDH_ECDSA_ENABLED
 	bool "ECDH-ECDSA based ciphersuite modes"
-	select MBEDTLS_ECDH_C
-	select MBEDTLS_ECDSA_C
+	depends on MBEDTLS_ECDH_C && MBEDTLS_ECDSA_C
 
 config MBEDTLS_ECDSA_DETERMINISTIC
 	bool "Deterministic ECDSA (RFC 6979)"
 
 config MBEDTLS_KEY_EXCHANGE_ECDH_RSA_ENABLED
 	bool "ECDH-RSA based ciphersuite modes"
-	select MBEDTLS_ECDH_C
+	depends on MBEDTLS_ECDH_C
 
 config MBEDTLS_KEY_EXCHANGE_ECJPAKE_ENABLED
 	bool "ECJPAKE based ciphersuite modes"
-	select MBEDTLS_ECJPAKE_C
+	depends on MBEDTLS_ECJPAKE_C
```

Comment:
during testing I discovered that this not only depends on `MBEDTLS_ECJPAKE_C` but also MBEDTLS_SHA256_C and MBEDTLS_ECP_DP_SECP256R1_ENABLED.

```
 * Requires: MBEDTLS_ECJPAKE_C
 *           MBEDTLS_SHA256_C
 *           MBEDTLS_ECP_DP_SECP256R1_ENABLED
```
shouldn't we reflect this in the Kconfig ?
Else users risk compilation errors during during build, but without proper help text guiding them to the root cause.

Errors seen:
```
```projects/github/ncs/modules/crypto/mbedtls/include/mbedtls/check_config.h:145:2: error: #error "MBEDTLS_ECP_C defined, but not all prerequisites"
  145 | #error "MBEDTLS_ECP_C defined, but not all prerequisites"
      |  ^~~~~
```
and
```
/projects/github/ncs/modules/crypto/mbedtls/include/mbedtls/ecp.h:343:2: error: #error "Missing definition of MBEDTLS_ECP_MAX_BITS"
  343 | #error "Missing definition of MBEDTLS_ECP_MAX_BITS"
      |  ^~~~~
```

Ref: https://github.com/zephyrproject-rtos/mbedtls/blob/7fed49c9b9f983ad6416986661ef637459723bcb/include/mbedtls/mbedtls_config.h#L946-L960

This error is already seen today on `zephyr/main`, but if changing the Kconfig, then addressing this issue also would be nice.